tools/swift-inspect/Sources/SwiftInspectLinux/PTrace.swift

```diff
@@ -0,0 +1,151 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import LinuxSystemHeaders
+
+public class PTrace {
+  enum PTraceError: Error {
+    case ptraceFailure(_ command: Int32, pid: pid_t, errno: Int32 = get_errno())
+    case waitFailure(pid: pid_t, errno: Int32 = get_errno())
+    case unexpectedWaitStatus(pid: pid_t, status: Int32, sigInfo: siginfo_t? = nil)
+  }
+
+  let pid: pid_t
+
+  // Initializing a PTrace instance attaches to the target process, waits for
+  // it to stop, and leaves it in a stopped state. The caller may resume the
+  // process by calling cont().
+  public init(process pid: pid_t) throws {
+    guard ptrace_attach(pid) != -1 else {
+      throw PTraceError.ptraceFailure(PTRACE_ATTACH, pid: pid)
+    }
+
+    while true {
+      var status: Int32 = 0
+      let result = waitpid(pid, &status, 0)
+      guard result != -1 else {
+        if get_errno() == EINTR { continue }
+        throw PTraceError.waitFailure(pid: pid)
+      }
+
+      precondition(pid == result, "waitpid returned unexpected value \(result)")
+
+      if wIfStopped(status) { break }
+    }
+
+    self.pid = pid
+  }
+
+  deinit { _ = ptrace_detach(self.pid) }
+
+  public func cont() throws {
```

Comment:
I think that we can spell out `continue` here. You can use ` to escape the keyword.